lib/gtfs/actions/general.js

```diff
@@ -1,107 +1,131 @@
+          console.warn(`Skipping GTFS entity request for ${agencyId}. Feed either not found or unpublished.`, activeProject)
+        }
+        continue
       }
-      if (routeId.indexOf(e.entity.RouteId) === -1) {
-        routeId.push(e.entity.RouteId)
+      const {publishedVersionId: namespace} = feed
+      // Is this the right feed ID for the GraphQL query?
+      const entityList = entitiesPerFeed[feed.id] || {namespace, routeId: [], stopId: []}
+      if (routeId !== null && entityList.routeId.indexOf(routeId) === -1) {
+        entityList.routeId.push(routeId)
       }
-      if (stopId.indexOf(e.entity.StopId) === -1) {
-        stopId.push(e.entity.StopId)
+      if (stopId !== null && entityList.stopId.indexOf(stopId) === -1) {
+        entityList.stopId.push(stopId)
       }
-    })
-    const method = 'post'
-    const body = JSON.stringify({
-      query: stopsAndRoutes(feedId, routeId, stopId),
-      variables: JSON.stringify({feedId, routeId, stopId})
-    })
-    dispatch(requestStopsAndRoutes(feedId, routeId, stopId, module))
-    return dispatch(secureFetch(GTFS_GRAPHQL_PREFIX, {method, body}))
-      .then(response => response.json())
-      .then(results => dispatch(receivedStopsAndRoutes(results, module)))
+      entitiesPerFeed[feed.id] = entityList
+    }
+    const entityRequests = Object.keys(entitiesPerFeed)
+      .map(feedId => {
+        const {namespace, routeId, stopId} = entitiesPerFeed[feedId]
+        dispatch(requestStopsAndRoutes(feedId, routeId, stopId, module))
+        return dispatch(fetchGraphQL({
+          query: stopsAndRoutes(namespace, routeId, stopId),
+          variables: {namespace, routeId, stopId}
+        }))
+          .then(data => dispatch(receivedStopsAndRoutes({feedId, results: data, moduleType})))
+      })
+    return Promise.all(entityRequests)
+    // .then(results => dispatch(receivedStopsAndRoutes(results, moduleType)))
   }
 }
 
-export function refreshGtfsElements (feedId, entities) {
-  return function (dispatch, getState) {
-    dispatch(requestGtfsElements(feedId, entities))
-    const bounds = getState().gtfs.filter.map.bounds
-    // if (!bounds) return
+function getStopsRoutesForBounds (feed: Feed, bounds: any, entities) {
+  return function (dispatch: dispatchFn, getState: getStateFn) {
     const maxLat = bounds.getNorth()
     const maxLon = bounds.getEast()
     const minLat = bounds.getSouth()
     const minLon = bounds.getWest()
-    const vars = {
-      feedId,
-      max_lat: maxLat,
-      max_lon: maxLon,
-      min_lat: minLat,
-      min_lon: minLon
-    }
-    return dispatch(secureFetch(compose(patternsAndStopsForBoundingBox(feedId, entities, maxLat, maxLon, minLat, minLon), vars)))
-      .then(response => response.json())
-      .then(results => {
-        return dispatch(receivedGtfsElements(feedId, results.stops, results.patterns))
+    const namespace = feed.publishedVersionId
+    // FIXME: replace with secure fetch
```

Comment:
let's maintain good security practices and implement secure fetch now.

Reply:
The `fetchGraphQL` action it dispatches uses secure fetch.  The comment should be removed.

Reply:
ok